examples/vision/aodnet.py

```diff
@@ -0,0 +1,376 @@
+            kernel_initializer=tf.initializers.random_normal(stddev=stddev),
+            kernel_regularizer=tf.keras.regularizers.L2(weight_decay),
+        )
+        self.conv_layer_2 = tf.keras.layers.Conv2D(
+            filters=3,
+            kernel_size=1,
+            strides=1,
+            padding="same",
+            activation=tf.nn.relu,
+            use_bias=True,
+            kernel_initializer=tf.initializers.random_normal(stddev=stddev),
+            kernel_regularizer=tf.keras.regularizers.L2(weight_decay),
+        )
+        self.conv_layer_3 = tf.keras.layers.Conv2D(
+            filters=3,
+            kernel_size=5,
+            strides=1,
+            padding="same",
+            activation=tf.nn.relu,
+            use_bias=True,
+            kernel_initializer=tf.initializers.random_normal(stddev=stddev),
+            kernel_regularizer=tf.keras.regularizers.L2(weight_decay),
+        )
+        self.conv_layer_4 = tf.keras.layers.Conv2D(
+            filters=3,
+            kernel_size=7,
+            strides=1,
+            padding="same",
+            activation=tf.nn.relu,
+            use_bias=True,
+            kernel_initializer=tf.initializers.random_normal(stddev=stddev),
+            kernel_regularizer=tf.keras.regularizers.L2(weight_decay),
+        )
+        self.conv_layer_5 = tf.keras.layers.Conv2D(
+            filters=3,
+            kernel_size=3,
+            strides=1,
+            padding="same",
+            activation=tf.nn.relu,
+            use_bias=True,
+            kernel_initializer=tf.initializers.random_normal(stddev=stddev),
+            kernel_regularizer=tf.keras.regularizers.L2(weight_decay),
+        )
+        self.relu = tf.keras.layers.ReLU(max_value=1.0)
+
+    def call(self, inputs, *args, **kwargs):
+        conv_1 = self.conv_layer_1(inputs)
+        conv_2 = self.conv_layer_2(conv_1)
+        concat_1 = tf.concat([conv_1, conv_2], axis=-1)
+        conv_3 = self.conv_layer_3(concat_1)
+        concat_2 = tf.concat([conv_2, conv_3], axis=-1)
+        conv_4 = self.conv_layer_4(concat_2)
+        concat_3 = tf.concat([conv_1, conv_2, conv_3, conv_4], axis=-1)
+        k = self.conv_layer_5(concat_3)
+        j = tf.math.multiply(k, inputs) - k + 1.0
+        output = self.relu(j)
+        return output
+
+
+"""
+Since we are training the model using a single GPU, i.e, Nvidia Tesla P100, we would use
+`tf.distribute.OneDeviceStrategy`, using this strategy will place any variables created
+in its scope on the specified device. We would define and compile our model in the scope
+of our distribution strategy. We would be using Adam as out optimizer, Mean Squared Error
+as the loss function and PSNR (Peak Signal Noise Ratio) as our metric.
+
+**Note:** For training AODNet in a multi-GPU environment,
+`tf.distribute.MirroredStrategy` can be used as a distribution stratgey.
+"""
+
+
+def peak_signal_noise_ratio(y_true, y_pred):
+    return tf.image.psnr(y_pred, y_true, max_val=255.0)
```

Comment:
This quantity is symmetric, no? Prefer using the standard signature `y_true, y_pred`